server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2130,7 +2129,7 @@ private VMTemplateVO updateTemplateOrIso(BaseUpdateTemplateOrIsoCmd cmd) {
             return template;
         }
 
-        template = _tmpltDao.createForUpdate(id);
+        template = _tmpltDao.findById(id);
```

Comment:
why is this changed?

Reply:
createForUpdate creates an empty object with just the id https://github.com/apache/cloudstack/blob/master/framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java#L295 , whereas a template object read from the db needed to check whether it's cross zone

Reply:
Line 2075 already has that, we need to check why createForUpdate was used in the first place.